### PR TITLE
Prevent incorrect error on simultaneous first wiki image request

### PR DESCRIPTION
**Pull request by nanaya**
Wait up to 10 seconds on first image load if other worker is already doing it and holding the lock.

### CODE REVIEW

**Comment by peppy** [comment]
Do you think this is the reason for the full failure earlier? Or something you found along the way?

**Comment by nanaya** [comment]
This probably is the reason.

**Comment by peppy** [comment]
I guess it's gotten worse due to now having 30+ workers.

File: app/Models/Wiki/Image.php [hunk 1]
```diff
@@ -67,10 +68,23 @@ public function sync($force = false)
...
 
         if (!$lock->get()) {
-            return $this;
+            if ($this->cache !== null) {
+                return $this;
+            }
+
+            try {
+                $lock->block(10);
+                $lock->release();
```
Comment (notbakaneko) [inline app/Models/Wiki/Image.php:81]: should this be in a `finally`?

**Review by notbakaneko — CHANGES_REQUESTED** [review]

File: app/Models/Wiki/Image.php (hunk 1 above)
Comment (nanaya) [reply app/Models/Wiki/Image.php:81]: I don't think so because in other cases the lock isn't obtained

Comment (notbakaneko) [reply app/Models/Wiki/Image.php:81]: 🤔 
(assuming the follow lines don't explode)

**Review by notbakaneko — DISMISSED** [review]

**Review by notbakaneko — APPROVED** [review]
apparently the whole fetching back can be done with `->lock(..)->block(...,fn () => ...) instead  🤔  (but I don't want to check everything again right now :eyes:)

File: app/Models/Wiki/Image.php (hunk 1 above)
Comment (nanaya) [reply app/Models/Wiki/Image.php:82]: actually this can't even be in finally as otherwise the lock would still be held when calling sync for return

**Review by notbakaneko — APPROVED** [review]

**Pull request merged**